Kudu.Services.Web/Env.cshtml

```diff
@@ -1,95 +1,103 @@
-    <ul class="fixed-width">
-        @foreach (DictionaryEntry entry in Environment.GetEnvironmentVariables().OfType<DictionaryEntry>().OrderBy(e => e.Key))
-         {
-            <li>@entry.Key = @entry.Value</li>
-         }
-    </ul>
-
-    <h3 id="path">PATH</h3>
-    <ul class="fixed-width">
-        @foreach (string folder in Environment.GetEnvironmentVariable("PATH").Trim(';').Split(';').OrderBy(s => s))
-         {
-            <li>@folder</li>
-         }
-    </ul>
-
-    <h3 id="httpHeaders">HTTP headers</h3>
-    <ul class="fixed-width">
-        @foreach (string name in Request.Headers.OfType<string>().OrderBy(s => s))
-         {
-            <li>@name=@Request.Headers[name]</li>
-         }
-    </ul>
-
-    <h3 id="serverVar">Server variables</h3>
-    <ul class="fixed-width">
-        @foreach (string name in Request.ServerVariables.OfType<string>().OrderBy(s => s))
-         {
-            <li>@name=@Request.ServerVariables[name]</li>
-         }
-    </ul>
+﻿@using System.Configuration
+@using System.Collections
+
+@{
+    Layout = "~/_Layout.cshtml";
+    Page.Title = "Environment";
+    var context = new HttpContextWrapper(HttpContext.Current);
+}
+
+<div class="container">
+    <h3>Index</h3>
+    <ul>
+        <li><a href="#sysInfo">System Info</a></li>
+        <li><a href="#appSettings">App Settings</a></li>
+        <li><a href="#connectionStrings">Connection Strings</a></li>
+        <li><a href="#envVariables">Environment variables</a></li>
+        <li><a href="#path">PATH</a></li>
+        <li><a href="#httpHeaders">HTTP Headers</a></li>
+        <li><a href="#serverVar">Server variables</a></li>
+    </ul>
+</div>
+
+<div class="container">
+    <h3 id="sysInfo">System info</h3>
+    <ul>
+        <li>System up time: @TimeSpan.FromMilliseconds(Environment.TickCount)</li>
+        <li>OS version: @Environment.OSVersion</li>
+        <li>64 bit system: @Environment.Is64BitOperatingSystem</li>
+        <li>64 bit process: @Environment.Is64BitProcess</li>
+        <li>Processor count: @Environment.ProcessorCount</li>
+        <li>Machine name: @Environment.MachineName</li>
+        <li>Instance id: @Kudu.Core.Infrastructure.InstanceIdUtility.GetInstanceId()</li>
+        <li>Short instance id: @Kudu.Core.Infrastructure.InstanceIdUtility.GetShortInstanceId()</li>
+        <li>CLR version: @Environment.Version</li>
+        <li>System directory: @Environment.SystemDirectory</li>
+        <li>Current working directory: @Environment.CurrentDirectory</li>
+        <li>IIS command line: @Environment.CommandLine</li>
+        @{
+            var homePath = Environment.GetEnvironmentVariable("HOME");
+            <li>@homePath usage: @Html.Raw(Kudu.Core.Environment.GetFreeSpaceHtml(homePath))</li>
+            if (Kudu.Core.Environment.IsAzureEnvironment())
+            {
+                <li>d:\local usage: @Html.Raw(@Kudu.Core.Environment.GetFreeSpaceHtml("d:\\local"))</li>

```

Comment:
Not exactly d:\local.   I think it is ok to hardcode given we check for Azure env.   I will robust the codes to check for Directory.Exists in addition.